pkgs/applications/graphics/apitrace/default.nix

```diff
@@ -15,7 +15,31 @@ stdenv.mkDerivation rec {
   # of games -- so it's fine to use e.g. bundled snappy.
   buildInputs = [ libX11 procps python2 libdwarf qtbase qtwebkit ];
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [ cmake wrapQtAppsHook ];
+
+  # Don't automatically wrap all binaries, I prefer to explicitly only wrap
+  # `qapitrace`.
+  dontWrapQtApps = true;
+
+  postFixup = ''
+    # Apitrace wraps some program. For some reason, they are not able to find
+    # the OpenGL driver anymore.
+    # We add back the position of the driver to the library search path.
+    # Adding the path for path 32 and 64 bits have no impact on the closure and
+    # is simpler because at build time of api trace, we don't know if the
+    # traced program will be 64 or 32 bits.
+    wrapProgram $out/bin/apitrace \
+      --prefix LD_LIBRARY_PATH : "/run/opengl-driver/lib:/run/opengl-driver-32/lib"
+
+    # {e}glretrace dynamically open the OpenGL driver at runtime, so they need
+    # a reference to libglvnd and libGL.
+    wrapProgram $out/bin/eglretrace \
+      --prefix LD_LIBRARY_PATH : "${stdenv.lib.makeLibraryPath [libglvnd libGL]}"
+    wrapProgram $out/bin/glretrace \
+      --prefix LD_LIBRARY_PATH : "${stdenv.lib.makeLibraryPath [libglvnd libGL]}"
```

Comment:
The problem with these binaries is that they don't have the library in their `NEEDED` ELF field, so we can't set the RPATH automatically

However, you should be able to use something like this to extend the rpath in `postFixup`:

for f in $out/bin/*; do
  patchelf --set-rpath "${stdenv.lib.makeLibraryPath [ libglvnd libGL ]}:$(patchelf --print-rpath $f)" $f 
done

Reply:
Indeed. Your solution also need to `chmod +x` before calling `patchelf`. I don't know if this is better or worse than the wrapper. Do you want me to change the PR to use your proposal?

Reply:
RPATH is usually better as it does not propagated to child processes.

Reply:
Good point. I'll fix that.

Reply:
This is now using `patchelf` and `--set-rpath` for everything.